src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/LambdaCompiler.Unary.cs

```diff
@@ -336,7 +336,7 @@ private void EmitConvert(UnaryExpression node, CompilationFlags flags)
                     UnaryExpression e = Expression.Convert(
                         Expression.Call(
                             node.Method,
-                            Expression.Convert(node.Operand, pis[0].ParameterType)
+                            Expression.Convert(node.Operand, paramType)
```

Comment:
This would seem to be why `paramType` was set above, and its not being used here a simple slip.